crypto/fipsmodule/bcm.c

```diff
@@ -342,3 +342,31 @@ void BORINGSSL_FIPS_abort(void) {
 }
 
 #endif  // BORINGSSL_FIPS
+
+#if !defined(BORINGSSL_FIPS) && !defined(BORINGSSL_SHARED_LIBRARY)
+// When linking with a static library, if no symbols in an object file are
+// referenced then the object file is discarded, even if it has constructor
+// function. For example, if an application is linking with libcrypto.a and
+// not referencing any symbol from crypto.o file, then crypto.o will be
+// discarded. This is an issue because we define the library constructor
+// in crypto.o so if the file is discarded then the library is not initialized.
+// Note that this is not a problem for the FIPS build because when building the
+// FIPS mode we have to ensure the initialization is done before the power-on
+// self-tests so the test function itself calls |OPENSSL_cpuid_setup|.
+//
+// So, the issue manifests only in the static non-FIPS build. The work around
+// it we add a dummy function |dummy_func_for_constructor| in |bcm.c| that
+// references a symbol, |CRYPTO_library_init| from |crypto.c|. This will ensure
```

Comment:
```suggestion
// calls the initialisation function |CRYPTO_library_init| defined in |crypto.c|. This will ensure
```
Same suggestion applies to the comment in internal.h, where I suggest to keep the comment and reference it here and in cpucap.c or wherever you prefer, but I think one copy is sufficient.

Reply:
fixed this. Removed the comment in internal.h and added a reference to the one in bcm.c